tests/mock_tests/fpmsyncd/receive_srv6_localsids_ut.cpp

```diff
@@ -0,0 +1,1203 @@
+        }
+
+        virtual void TearDown() override
+        {
+        }
+    };
+}
+
+namespace ut_fpmsyncd
+{
+    /* Test Receiving an SRv6 Local SID nexthop bound to the End behavior */
+    TEST_F(FpmSyncdSRv6LocalSIDsTest, RecevingRouteWithSRv6LocalSIDEnd)
+    {
+        ASSERT_NE(m_routeSync, nullptr);
+
+        /* Create a Netlink object containing an SRv6 Local SID */
+        IpAddress _localsid = IpAddress("fc00:0:1:1::");
+        int8_t _block_len = 32;
+        int8_t _node_len = 16;
+        int8_t _func_len = 16;
+        int8_t _arg_len = 0;
+        uint32_t _action = SRV6_LOCALSID_ACTION_END;
+
+        struct nlmsg *nl_obj = create_srv6_localsid_nlmsg(RTM_NEWSRV6LOCALSID, &_localsid, _block_len, _node_len, _func_len, _arg_len, _action);
+        if (!nl_obj)
+            throw std::runtime_error("SRv6 Local SID creation failed");
+
+        /* Send the Netlink object to the FpmLink */
+        m_fpmLink->processRawMsg(&nl_obj->n);
+
+        /* Check that fpmsyncd created the correct entries in APP_DB */
+        std::string action;
+        ASSERT_EQ(m_srv6LocalSidTable->hget("32:16:16:0:fc00:0:1:1::", "action", action), true);
+        ASSERT_EQ(action, "end");
+
+        /* Destroy the Netlink object and free the memory */
+        free_nlobj(nl_obj);
+
+        /* Delete Local SID */
+        nl_obj = create_srv6_localsid_nlmsg(RTM_DELSRV6LOCALSID, &_localsid, _block_len, _node_len, _func_len, _arg_len, _action);
+        if (!nl_obj)
+            throw std::runtime_error("SRv6 Local SID creation failed");
+
+        /* Send the Netlink object to the FpmLink */
+        m_fpmLink->processRawMsg(&nl_obj->n);
+
+        /* Check that fpmsyncd created the correct entries in APP_DB */
+        ASSERT_EQ(m_srv6LocalSidTable->hget("32:16:16:0:fc00:0:1:1::", "action", action), false);
+
+        /* Destroy the Netlink object and free the memory */
+        free_nlobj(nl_obj);
+    }
+
+    /* Test Receiving an SRv6 Local SID nexthop bound to the End.X behavior */
+    TEST_F(FpmSyncdSRv6LocalSIDsTest, RecevingRouteWithSRv6LocalSIDEndX)
+    {
+        ASSERT_NE(m_routeSync, nullptr);
+
+        /* Create a Netlink object containing an SRv6 Local SID */
+        IpAddress _localsid = IpAddress("fc00:0:1:1::");
+        int8_t _block_len = 32;
+        int8_t _node_len = 16;
+        int8_t _func_len = 16;
+        int8_t _arg_len = 0;
+        uint32_t _action = SRV6_LOCALSID_ACTION_END_X;
+        IpAddress _adj = IpAddress("2001:db8:1::1");
+
+        struct nlmsg *nl_obj = create_srv6_localsid_nlmsg(RTM_NEWSRV6LOCALSID, &_localsid, _block_len, _node_len, _func_len, _arg_len, _action, NULL, &_adj);
+        if (!nl_obj)
+            throw std::runtime_error("SRv6 Local SID creation failed");
+
+        /* Send the Netlink object to the FpmLink */
+        m_fpmLink->processRawMsg(&nl_obj->n);
+
+        /* Check that fpmsyncd created the correct entries in APP_DB */
+        std::string action;
```

Comment:
Could you pls create a wrapper to create MYSID table key by passing block/node/func/arg lengths and local SID? That will be very useful as this is used in many places.

Reply:
Done. I added a function `get_srv6_my_sid_table_key()` that builds the MYSID table key and change the code to use this function.